### PR TITLE
UPSTREAM: 75785: add annotation from namespace controller to indicate…

### PR DESCRIPTION
**Pull request by deads2k**
… why deletion failed

/assign @mfojtik

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/controller/namespace/deletion/namespaced_resources_deleter.go [hunk 1]
```diff
@@ -295,6 +298,20 @@ func (d *namespacedResourcesDeleter) updateNamespaceStatusFunc(namespace *v1.Nam
 	return d.nsClient.UpdateStatus(&newNamespace)
 }
 
+// updateSetDeletionFailureFunc will set an annotation indicating that the namespace controller has failed to delete all content
+func (d *namespacedResourcesDeleter) updateSetDeletionFailureFunc(err error) func(namespace *v1.Namespace) (*v1.Namespace, error) {
+	errorAnnotationName := "namespace-controller.kube-controller-manager.kubernetes.io/deletion-error"
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/pkg/controller/namespace/deletion/namespaced_resources_deleter.go]: before merge, I need to update this to be an alpha something in openshift.io for when the real status pull gets merged upstream

**Comment by mfojtik** [comment]
neat

/lgtm

**Comment by deads2k** [comment]
makes debugging ns cleanup possible

**Pull request merged**